Tasks/AzureTestPlanV0/testLibExecutor.ts

```diff
@@ -0,0 +1,91 @@
+import os = require('os');
+import Q = require('q');
+import path = require('path');
+import fs = require('fs');
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as tr from 'azure-pipelines-task-lib/toolrunner';
+
+var isWindows = os.type().match(/^Win/);
+var m2HomeEnvVar: string = null;
+var mavenPOMFile: string = tl.getPathInput('mavenPOMFile', true, true);
+
+function getMavenExec(){
+    var mvnExec: string = '';
+    m2HomeEnvVar = tl.getVariable('M2_HOME');
+    if (m2HomeEnvVar) {
+        tl.debug('Using M2_HOME environment variable value for Maven path: ' + m2HomeEnvVar);
+        mvnExec = path.join(m2HomeEnvVar, 'bin', 'mvn');
+    }
+    // Second, look for Maven in the system path
+    else {
+        tl.debug('M2_HOME environment variable is not set, so Maven will be sought in the system path');
+        mvnExec = tl.which('mvn', true);
+    }
+
+    if (isWindows &&
+        !mvnExec.toLowerCase().endsWith('.cmd') &&
+        !mvnExec.toLowerCase().endsWith('.bat')) {
+        if (tl.exist(mvnExec + '.cmd')) {
+            // Maven 3 uses mvn.cmd
+            mvnExec += '.cmd';
+        }
+        else if (tl.exist(mvnExec + '.bat')) {
+            // Maven 2 uses mvn.bat
+            mvnExec += '.bat';
+        }
+    }
+
+    tl.checkPath(mvnExec, 'maven path');
```

Comment:
Just for my understanding, what does this checkPath do?

Reply:
If the path is not found, safeguard to exit since checkPath function provides the ability to check if the path does not exist, the task will fail with an error message. Execution will halt.